hack/util.sh

```diff
@@ -632,7 +637,18 @@ function install_router {
 function install_registry {
 	# The --mount-host option is provided to reuse local storage.
 	echo "[INFO] Installing the registry"
-	openshift admin registry --config="${ADMIN_KUBECONFIG}" --images="${USE_IMAGES}"
+	# For testing purposes, ensure the quota objects are always up to date in the registry by
+	# disabling project cache.
+	openshift admin registry --config="${ADMIN_KUBECONFIG}" --images="${USE_IMAGES}" -o json | \
+		sed '/"name"\s*:\s*"REGISTRY_HTTP_ADDR"/ i\
+			"name": "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_PROJECTCACHETTL", "value": "0"},{' | \
```

Comment:
oadm registry ... | oc set env -f - .... | oc create -f -

On Mon, Apr 11, 2016 at 2:44 PM, Michal Minar <notifications@github.com>
wrote:

> In hack/util.sh
> <https://github.com/openshift/origin/pull/8195#discussion_r59258612>:
>
> > @@ -632,7 +637,18 @@ function install_router {
> >  function install_registry {
> >  	# The --mount-host option is provided to reuse local storage.
> >  	echo "[INFO] Installing the registry"
> > -	openshift admin registry --config="${ADMIN_KUBECONFIG}" --images="${USE_IMAGES}"
> > +	# For testing purposes, ensure the quota objects are always up to date in the registry by
> > +	# disabling project cache.
> > +	openshift admin registry --config="${ADMIN_KUBECONFIG}" --images="${USE_IMAGES}" -o json | \
> > +		sed '/"name"\s*:\s*"REGISTRY_HTTP_ADDR"/ i\
> > +			"name": "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_PROJECTCACHETTL", "value": "0"},{' | \
>
> Already tried that. But there are places like oc logs
> <https://github.com/miminar/origin/blob/master/test/end-to-end/core.sh#L89>
> in test suite that expect the registry to be deployed in the first version
> of dc and I didn't want to touch them. And it was unstable (dc-2 failed to
> scale sometimes) and I was too lazy to debug.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8195/files/a0585fd77f099f1089f4964d7b672a3f79509c7c#r59258612>
>